extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlCreate.java

```diff
@@ -65,4 +74,119 @@ public ApiType execute() throws KubectlException {
       }
     }
   }
+  /**
+   * Loads an API object from a YAML string representation without knowing the object type
+   * and submits any type of resource.
+   *
+   * Note: This method works for single-instance YAML files only.
+   * If the YAML contains multiple resources, it will fail.
+   * Support for multiple resources in a single YAML file is not implemented yet.
+   *
+   * @param content The YAML content as a Reader
+   * @return The created resource object
+   * @throws Exception If there is an error in parsing and submitting resources.
+   */
+  public static Object loadAndSubmitResource(Reader content) throws Exception {
+    // Initialize the API client and Discovery instance
+    ApiClient client = Configuration.getDefaultApiClient();
+    Discovery discovery = new Discovery(client);
+
+    // Load the YAML content as an unstructured object
+    Object unstructuredObject = Yaml.load(content);
+
+    // Ensure the unstructured object is a Map
+    if (!(unstructuredObject instanceof Map)) {
+      throw new ParseException("Invalid YAML", 0);
+    }
+
+    // Typecasting the unstructured object to a Map
+    Map<String, Object> yamlMap = (Map<String, Object>) unstructuredObject;
+
+    // Getting apiVersion and kind
+    String apiVersion = (String) yamlMap.get("apiVersion");
+    String kind = (String) yamlMap.get("kind");
+
+    // Validate apiVersion and kind
+    if (apiVersion == null || kind == null) {
+      throw new ParseException("YAML does not contain apiVersion or kind", 0);
+    }
+
+    // Finding the resource in the discovery class
+    Set<Discovery.APIResource> resources = discovery.findAll();
+    Optional<Discovery.APIResource> apiResource = resources.stream()
+            .filter(r -> r.getKind().equalsIgnoreCase(kind))
```

Comment:
You need to check group/version here also. There are a few Kinds which are duplicates in several different api groups :(

Reply:
See https://github.com/kubernetes-client/java/issues/3668#issuecomment-2324638774

Core.V1Event vs Events.V1Event